File: h5gl/new/src/pan3d/skill/key/SkillEffect.ts
module Pan3d {
    export class SkillEffect extends SkillKey {
        public active: Object3D;

        public addToRender(): void {
            super.addToRender();
            this.particle.addEventListener(BaseEvent.COMPLETE, this.onPlayCom, this);
        }

        protected onPlayCom(event: Event = null): void {
            this.particle.removeEventListener(BaseEvent.COMPLETE, this.onPlayCom, this);
            this.scene3D.particleManager.removeParticle(this.particle);
            this.removeCallFun(this);
        }


    }
}

File: android/app/src/main/java/com/pan3d/skill/key/SkillEffect.java
package com.pan3d.skill.key;

import com.pan3d.base.Object3D;
import com.pan3d.event.BaseEvent;
import com.pan3d.event.EventCallBack;
import com.pan3d.filemodel.ParticleManager;
import com.pan3d.scene.Scene3D;

public class SkillEffect extends SkillKey{

    private static final String TAG = "SkillEffect";
    public Object3D active;

    public SkillEffect(Scene3D val) {
        super(val);
    }


    public void addToRender(ParticleManager val) {
        super.addToRender(val);
         this.particle.addEventListener(BaseEvent.COMPLETE,this.getOnPlayCom() , this);
    }
    EventCallBack onPlayCom;
    private EventCallBack getOnPlayCom(){
        if(onPlayCom!=null){
            return onPlayCom;
        }
        onPlayCom=new EventCallBack() {
            @Override
            public void call(Object val, BaseEvent event) {
               particle.removeEventListener(BaseEvent.COMPLETE,getOnPlayCom(),this);
                scene3D.particleManager.removeParticle(particle);
                removeCallFun.StateChange(this);
            }
        };
        return onPlayCom;
    }


}
